PWGJE/Tasks/jetChCorr.cxx

```diff
@@ -0,0 +1,439 @@
+    registry.add("hr3_ch_d_zg", ";#it{z}_{g} (GeV/#it{c}) {nsd3};N_{ch} (h#bar{h})", {HistType::kTH1F, {{20, 0., .5}}});
+    registry.add("hr3_ch_s_rg", ";#it{r}_{g} (GeV/#it{c}) {nsd3};N_{ch} (hh)", {HistType::kTH1F, {{20, -2., 0.}}});
+    registry.add("hr3_ch_d_rg", ";#it{r}_{g} (GeV/#it{c}) {nsd3};N_{ch} (h#bar{h})", {HistType::kTH1F, {{20, -2., 0.}}});
+    registry.add("hr3_ch_s_lu", ";#it{z}_{g} {nsd3};#it{r}_{g}", {HistType::kTH2F, {{20, 0., 5.}, {28, -2., 5.}}});
+    registry.add("hr3_ch_d_lu", ";#it{z}_{g} {nsd3};#it{r}_{g}", {HistType::kTH2F, {{20, 0., 5.}, {28, -2., 5.}}});
+
+    jetReclusterer.isReclustering = true;
+    jetReclusterer.algorithm = fastjet::JetAlgorithm::cambridge_algorithm;
+  }
+
+  // template <typename JetTable, typename JetTableMCP, typename SubstructureTable>
+  template <bool isMCP, bool isSubtracted, typename T, typename U>
+  void jetReclustering(T const& jet, U& outputTable)
+  {
+    jetReclustered.clear();
+    fastjet::ClusterSequenceArea clusterSeq(jetReclusterer.findJets(jetConstituents, jetReclustered));
+    jetReclustered = sorted_by_pt(jetReclustered);
+    fastjet::PseudoJet daughterSubJet = jetReclustered[0];
+    fastjet::PseudoJet parentSubJet1;
+    fastjet::PseudoJet parentSubJet2;
+    bool softDropped = false;
+    auto nsd = 0.0;
+    auto zg = -1.0;
+    auto rg = -1.0;
+    std::vector<float> energyMotherVec;
+    std::vector<float> ptLeadingVec;
+    std::vector<float> ptSubLeadingVec;
+    std::vector<float> thetaVec;
+
+    float ptJet = daughterSubJet.pt();
+
+    while (daughterSubJet.has_parents(parentSubJet1, parentSubJet2)) {
+      if (parentSubJet1.perp() < parentSubJet2.perp()) {
+        std::swap(parentSubJet1, parentSubJet2);
+      }
+
+      vector<fastjet::PseudoJet> constituents1 = parentSubJet1.constituents();
+      vector<fastjet::PseudoJet> constituents2 = parentSubJet2.constituents();
+      // cout<<"in subJET1 ********************************************* "<<endl;
+      int found1 = 0;
+      int found2 = 0;
+      for (unsigned int j = 0; j < constituents1.size(); j++) {
+        // cout<<constituents1[j].user_index()-1<<", ";
+        if ((n_trackL == (constituents1[j].user_index() - 1)) || (trackL == (constituents1[j].user_index() - 1)))
+          found1++;
+      }
+      // cout<<endl;
+      // cout<<"in subJET2 ********************************************* "<<endl;
+      for (unsigned int j = 0; j < constituents2.size(); j++) {
+        // cout<<constituents2[j].user_index()-1<<", ";
+        if ((n_trackL == constituents2[j].user_index() - 1) || (trackL == constituents2[j].user_index() - 1))
+          found2++;
+      }
+      // cout<<endl;
+
+      auto z = parentSubJet2.perp() / (parentSubJet1.perp() + parentSubJet2.perp());
+      auto theta = parentSubJet1.delta_R(parentSubJet2);
+      energyMotherVec.push_back(daughterSubJet.e());
```

Comment:
if you only work with histograms and not trees you can remove this and terminate the clustering at the step where the leading particles were found in opposite subjets

Reply:
terminated.